patches/tModLoader/Terraria/GameContent/Liquid/LiquidEdgeRenderer.cs

```diff
@@ -54,21 +55,33 @@ public static class LiquidEdgeRenderer
 		AlphaDestinationBlend = Blend.InverseSourceAlpha
 	};
 
-	private static List<Point> BlockWaterBehindLocations { get; } = [];
+	private struct EdgeSpan
+	{
+		public ushort X;
+		public ushort YStart;
+		public byte Height;
```

Comment:
Please make `ushort`, just for peace of mind (though I doubt render area ever goes over 4080px high). The struct size is gonna be the same regardless. 

Code might be ever-so-slightly cleaner if this was `ushort YEnd`.